### PR TITLE
Update contributing and testing guidelines, Rakefile and Travis configuration

### PR DESCRIPTION
**Pull request by legal90**
- Update contributing and testing guidelines according to Chef Softwars conventions
  https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD
  https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD
- Use ChefDK for running Rake tasks and Travis jobs

### CODE REVIEW

**Comment by legal90** [comment]
@patcon @racktear @Kasen  Please, review

**Comment by legal90** [comment]
This PR was inspired by these cookbooks:
https://github.com/chef-cookbooks/mysql
https://github.com/chef-cookbooks/database

**Comment by Kasen** [comment]
:+1:

**Comment by patcon** [comment]
:+1:

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,55 +1,54 @@
...
+the change or offer suggestions.
 
-I regularly review contributions and will get back to you if I have
-any suggestions or concerns.
+## The Apache License and the CLA/CCLA
+Licensing is very important to open source projects. It helps ensure the
+software continues to be available under the terms that the author desired.
 
-## The Apache License
+Parallels uses [the Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0)
```
Comment (knazarov) [inline CONTRIBUTING.md]: This is a very broad statement and should be avoided. Or if we really do so for all cookbooks, please state that explicitly.

File: Guardfile [hunk 2]
```diff
@@ -1,12 +1,17 @@
 # A sample Guardfile
```
Comment (knazarov) [inline Guardfile]: Just noticing: "A sample Guardfile"

File: TESTING.md [hunk 3]
```diff
@@ -1,47 +1,83 @@
...
+cloud drivers.
+
+```
+$ rake -T
+rake integration:vagrant      # Run Test Kitchen with Vagrant
+rake spec                     # Run ChefSpec examples
+rake style                    # Run all style checks
+rake style:chef               # Run Chef style checks
+rake style:ruby               # Run Ruby style checks
+rake style:ruby:auto_correct  # Auto-correct RuboCop offenses
```
Comment (knazarov) [inline TESTING.md:31]: Do you support auto_correct in the current Rakefile?

Comment (legal90) [inline TESTING.md:31]: Yes. It is defined in `rubocop/rake_task`
https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/rake_task.rb

**Pull request merged**